src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ReleaseProcessorService.cs

```diff
@@ -1,252 +0,0 @@
-using System;
-using System.Linq;
-using System.Threading.Tasks;
-using AutoMapper;
-using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
-using GovUk.Education.ExploreEducationStatistics.Data.Importer.Utils;
-using GovUk.Education.ExploreEducationStatistics.Data.Model;
-using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
-using GovUk.Education.ExploreEducationStatistics.Data.Processor.Model;
-using GovUk.Education.ExploreEducationStatistics.Data.Processor.Models;
-using GovUk.Education.ExploreEducationStatistics.Data.Processor.Services.Interfaces;
-using Microsoft.EntityFrameworkCore;
-using Microsoft.Extensions.Logging;
-using static GovUk.Education.ExploreEducationStatistics.Common.Model.FileType;
-using Publication = GovUk.Education.ExploreEducationStatistics.Data.Model.Publication;
-using Release = GovUk.Education.ExploreEducationStatistics.Data.Model.Release;
-using Theme = GovUk.Education.ExploreEducationStatistics.Data.Model.Theme;
-using Topic = GovUk.Education.ExploreEducationStatistics.Data.Model.Topic;
-
-namespace GovUk.Education.ExploreEducationStatistics.Data.Processor.Services
-{
-    public class ReleaseProcessorService : IReleaseProcessorService
-    {
-        private readonly IMapper _mapper;
-        private readonly ILogger<ReleaseProcessorService> _logger;
-
-        public ReleaseProcessorService(IMapper mapper, ILogger<ReleaseProcessorService> logger)
-        {
-            _mapper = mapper;
-            _logger = logger;
-        }
-
-        public Subject CreateOrUpdateRelease(SubjectData subjectData, ImportMessage message,
-            StatisticsDbContext statisticsDbContext, ContentDbContext contentDbContext)
-        {
-            var theme = CreateOrUpdateTheme(message, statisticsDbContext).Result;
-            var topic = CreateOrUpdateTopic(message, statisticsDbContext, theme).Result;
-            var publication = CreateOrUpdatePublication(message, statisticsDbContext, topic).Result;
-            var release = CreateOrUpdateRelease(message, statisticsDbContext, publication).Result;
-
-            RemoveSubjectIfExisting(message, statisticsDbContext);
```

Comment:
Are we now no longer needing to do this "remove existing subject" step in the new way of doing things?

Reply:
Discussed this on Slack. This probably to cover the case where the processor crashed or restarted after the subject was created. As the subject is created initially, this case no longer needs to be covered by the processor.